File: MIDTERM-LAB-3/MP04/MP04_CountRows.js
const fs = require('fs');
const readline = require('readline');

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

rl.question('Enter CSV file path: ', (filePath) => {
    try {
        const data = fs.readFileSync(filePath, 'utf8');
        // Split by newline and filter out empty strings
        const rows = data.split('\n').filter(row => row.trim() !== "");
        
        // Logic: Total rows minus 1 for the header
        const validCount = rows.length > 0 ? rows.length - 1 : 0;
        console.log(`Total valid data rows: ${validCount}`);
    } catch (err) {
        console.error("Error:", err.message);
    }
    rl.close();
});

File: MIDTERM-LAB-3/MP04/MP04_CountRows.java
package MP04;

import java.io.*;
import java.util.Scanner;

public class MP04_CountRows {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter CSV file path: ");
        String filePath = sc.nextLine();

        int rowCount = 0;
        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
            String line;
            br.readLine(); // Skip header row
            
            while ((line = br.readLine()) != null) {
                // Logic: Only count non-empty lines
                if (!line.trim().isEmpty()) {
                    rowCount++;
                }
            }
            System.out.println("Total valid data rows: " + rowCount);
        } catch (IOException e) {
            System.out.println("Error: " + e.getMessage());
             sc.close();
        }
    }
}
